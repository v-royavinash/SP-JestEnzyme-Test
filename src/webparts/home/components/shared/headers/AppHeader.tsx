import React from "react";
import Navbar from "react-bootstrap/Navbar";
import styles from "./Header.module.scss";

const AppHeader = () => {
    const mslogo: string = require("../../../assets/images/mslogo.png");

    return (
        <Navbar className={styles.navbg}>
            <Navbar.Brand
                href="#/"
                className={styles.bgDecorator}
            >
                <span className={styles.headerPosition}>
                    <span className={styles.appHeaderLabel}>Test</span>
                </span>
            </Navbar.Brand>
        </Navbar>
    );
};

export default AppHeader;
