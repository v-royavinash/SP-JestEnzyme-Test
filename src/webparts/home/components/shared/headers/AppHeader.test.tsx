/// <reference types="jest" />

import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import AppHeader from './AppHeader';

describe('Shared Component : AppHeader', () => {

    let reactComponent: ReactWrapper<{}, {}>;

    beforeEach(() => {
        reactComponent = mount(React.createElement(
            AppHeader, {}
        ));
    });

    afterEach(() => reactComponent.unmount());

    test('Check if there is a div element in the component rendering', () => {
        expect(reactComponent.find('div')).not.toBeNull();
    });

    test('Check if header title is as expected', () => {
        let appHeaderTitle = reactComponent.find('span').first().text();

        expect(appHeaderTitle).toBe("Test");
    });

    test('Check if the component matches the snapshot', () => {
        expect(reactComponent.html).toMatchSnapshot();
    });

});