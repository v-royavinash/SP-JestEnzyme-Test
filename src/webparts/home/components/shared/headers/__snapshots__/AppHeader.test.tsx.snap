// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Component : AppHeader Check if the component matches the snapshot 1`] = `[Function]`;
